src/components/calcite-radio-button/calcite-radio-button.tsx

```diff
@@ -123,26 +124,35 @@ export class CalciteRadioButton {
 
   private titleAttributeObserver: MutationObserver;
 
+  /** @internal */
+  @Method()
+  async emitCheckedChange() {
+    this.calciteRadioButtonCheckedChange.emit();
+  }
+
   //--------------------------------------------------------------------------
   //
   //  Private Methods
   //
   //--------------------------------------------------------------------------
 
-  private checkFirstRadioButton(): void {
+  private checkLastRadioButton(): void {
     const radioButtons = Array.from(document.querySelectorAll("calcite-radio-button")).filter(
```

Comment:
Instead of filtering after querying all radio buttons, would it be better to do this:

```
Array.from(document.querySelectorAll(`calcite-radio-button[name=${this.name}]`))
```

Reply:
I had lots of problems with that approach in places in the past.  For some reason those queries weren't always working right because of some attribute reflection issues.

Reply:
Here's one spot where I did that.  I think this was @jcfranco 's idea originally when he reviewed it https://github.com/Esri/calcite-components/pull/1072/files#diff-24f2982716ec00de0be7f0b9390bf88c1bd8ab7274a9c8e2d83db98d1a4cd7c4L125-R133

Reply:
Oh that's weird. Should we open a bug report?

Reply:
Probably?  Would want to setup a codepen or GH repo that repros it.  I can look into that.  Curious if I can repro it again.